### PR TITLE
Rename templates

### PR DESCRIPTION
**Pull request by melindaloubser1**
**Proposed changes**:
- Rename `templates` key in domain to `responses` for consistency across docs
- closes https://github.com/RasaHQ/roadmap/issues/400

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/core/actions.rst [hunk 1]
```diff
@@ -12,7 +12,7 @@ Actions
 Actions are the things your bot runs in response to user input.
 There are four kinds of actions in Rasa:
 
- 1. **Utterance actions**: start with ``utter_`` and send a specific message
+ 1. **Response actions**: start with ``utter_`` and send a specific message
```
Comment (melindaloubser1) [inline docs/core/actions.rst]: @amn41 Should utterance actions be renamed like this?

Comment (amn41) [reply docs/core/actions.rst]: hmm good question. Would be more consistent, but then we'd probably have to change the classname and the `utter_` prefix. I'd say we keep calling them utterance actions. wdyt @abhilasharoy @tmbo ?

Comment (abhilasharoy) [reply docs/core/actions.rst]: I agree that they should be consistent, but I also think that we need a bigger discussion to make sure any affected pieces are changed accordingly (like the `utter_` prefix that @amn41 mentioned). 

So for now, I'd also vote to keep it as "utterance actions".

Comment (melindaloubser1) [reply docs/core/actions.rst]: Alrighty, I have reverted it to utterance actions.

File: docs/core/domains.rst [hunk 2]
```diff
@@ -81,25 +81,25 @@ The ``name`` function of ``MyAwesomeAction`` needs to return
 ``my_custom_action`` in this example (for more details,
 see :ref:`custom-actions`).
 
-.. _utter_templates:
+.. _domain-responses:
 
-Utterance templates
+Responses
 -------------------
```
Comment (amn41) [inline docs/core/domains.rst]:
```suggestion
---------
```

File: docs/core/domains.rst [hunk 3]
```diff
@@ -186,19 +186,20 @@ can be sent like so:
                text: Select a date
 
 
-Channel-Specific Utterances
+Channel-Specific Responses
 ---------------------------
```
Comment (amn41) [inline docs/core/domains.rst]:
```suggestion
--------------------------
```

File: docs/core/domains.rst [hunk 4]
```diff
@@ -186,19 +186,20 @@ can be sent like so:
...
 
-If you have certain utterances that you would like sent only to specific
+For each response, you can have multiple **response templates** (see :ref:`variations`).
+If you have certain response templates that you would like sent only to specific
 channels, you can specify this with the ``channel:`` key. The value should match
 the name defined in the ``name()`` method of the channel's ``OutputChannel``
-class. Channel-specific utterances are especially useful if creating custom
+class. Channel-specific response templates are especially useful if creating custom
```
Comment (amn41) [inline docs/core/domains.rst]:
```suggestion
class. Channel-specific responses are especially useful if creating custom
```

File: docs/core/stories.rst [hunk 5]
```diff
@@ -82,19 +82,19 @@ entities (you can, however, change this behavior using the
...
+and custom actions. Utterance actions are hardcoded messages that a bot can respond
 with. Custom actions, on the other hand, involve custom code being executed.
 
-All actions (both utterances and custom actions) executed by the bot are shown
+All actions (both utterance actions and custom actions) executed by the bot are shown
 as lines starting with ``-`` followed by the name of the action.
 
-All utterances must begin with the prefix ``utter_``, and must match the name
-of the template defined in the domain.
+All utterance actions must begin with the prefix ``utter_``, and must match the name
```
Comment (amn41) [inline docs/core/stories.rst]:
```suggestion
The responses for utterance actions must begin with the prefix ``utter_``, and must match the name
```

File: docs/user-guide/validate-files.rst [hunk 6]
```diff
@@ -39,8 +39,8 @@ which has the following methods:
 
 **verify_intents_in_stories():** Verification for intents in the stories, to check if they are valid.
 
-**verify_utterances():** Checks domain file for consistency between utterance templates and utterances listed under
-actions.
+**verify_utterances():** Checks domain file for consistency between responses listed in the `responses` section 
+and responses listed under actions.
```
Comment (amn41) [inline docs/user-guide/validate-files.rst]:
```suggestion
and the utterance actions you have defined.
```

File: examples/moodbot/domain.yml [hunk 7]
```diff
@@ -7,7 +7,15 @@ intents:
   - mood_unhappy
   - bot_challenge
 
-templates:
+actions:
```
Comment (amn41) [inline examples/moodbot/domain.yml]: why had this section been added ?

File: rasa/core/domain.py [hunk 8]
```diff
@@ -123,7 +123,14 @@ def from_yaml(cls, yaml: Text) -> "Domain":
 
     @classmethod
     def from_dict(cls, data: Dict) -> "Domain":
-        utter_templates = cls.collect_templates(data.get("templates", {}))
+        utter_templates = cls.collect_templates(data.get("responses", {}))
+        if "templates" in data:
+            warnings.warn(
+                "'templates' in the domain should be renamed to 'responses'",
```
Comment (amn41) [inline rasa/core/domain.py]:
```suggestion
                "Your domain file contains the key: 'templates'. This has been deprecated and renamed to 'responses'. The 'templates' key will no longer work in future versions of Rasa. Please replace 'templates' with 'responses'",
```

File: .gitignore [hunk 9]
```diff
@@ -85,3 +85,4 @@ events.db
 env
 .dir-locals.el
 .history
+rasadev.code-workspace
```
Comment (amn41) [inline .gitignore]: did you mean to commit this?

Comment (melindaloubser1) [reply .gitignore]: nope, my mistake

File: examples/moodbot/domain.yml (hunk 7 above)
Comment (melindaloubser1) [reply examples/moodbot/domain.yml]: trying to figure out why it shows up that way - local diff only shows templates -> responses

Comment (amn41) [reply examples/moodbot/domain.yml]: could just be an error on github's rendering side

**Pull request merged**